Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -934,9 +944,38 @@ bool Renderer::StartFrameDumpToImage(const FrameDumpConfig& config)
   return true;
 }
 
+void writePPM(std::ostream& os, int width, int height, int stride, const u8* data)
+{
+  // header
+  os << "P6\n"
+     << width << " " << height << "\n"
+     << "255\n";
+
+  // pixels. we omit the alpha channel
+  for (int row = 0; row < height; ++row)
+  {
+    int index = row * stride;
+    for (int col = 0; col < width; ++col)
+    {
+      os.write(reinterpret_cast<const char*>(data) + index, 3);
+      index += 4;
+    }
+  }
+}
+
 void Renderer::DumpFrameToImage(const FrameDumpConfig& config)
 {
   std::string filename = GetFrameDumpNextImageFileName();
-  TextureToPng(config.data, config.stride, filename, config.width, config.height, false);
+  if (g_Config.bDumpFramesToPPM)
+  {
+    std::ofstream out;
```

Comment:
You can just do `std::ofstream out{filename}`. Instead of using an `open()` call.

`close()` can also be omitted, as this will automatically be done when the scope is exited.